File: python/P7.py
class Solution(object):
    def reverse(self, x):
        string = str(x)
        if string[0] == '-':
            new_string = ''.join(reversed(string[1:]))
            return -int(new_string) if -int(new_string)>-2**31-1 else 0 
        else:
            new_string = string[::-1]
            return int(new_string) if int(new_string)<2**31-1 else 0
        

File: java/P7.java
class Solution {
    public int reverse(int x) {
        long answer = 0;
        while (x!=0) {
            answer = answer*10+x%10;
            x /=10;
        }
        return (answer>Integer.MAX_VALUE || answer<Integer.MIN_VALUE) ? 0: (int)answer;
    }
}